packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -106,6 +107,23 @@ class DataSourceWithBackend<
     super(instanceSettings);
   }
 
+  getRequestUrl = (request: DataQueryRequest): string => {
+    const pubdashEnabled = getConfig().featureToggles.publicDashboards;
+    const publicView = getConfig().isPublicDashboardView;
+
+    // Assuming url /public-dashboards/:uid
+    const path = window.location.pathname;
```

Comment:
@joshhunt I'm assuming Redux would be the best way to save the current public dashboard uid? The url props get passed into the DashboardPage component, but thats super far away from here.

Reply:
you can call getDashboardSrv().getCurrent() to get the current dashboard, and it should have the public dashboard uid (in meta or in main uid).